packages/compass-aggregations/src/modules/saved-pipeline.ts

```diff
@@ -0,0 +1,168 @@
+  return { ...state, pipelines: action.pipelines, isLoaded: true };
+};
+
+const MAPPINGS = {
+  [SET_SHOW_SAVED_PIPELINES]: setShowSavedPipelinesList,
+  [SAVED_PIPELINE_ADD]: addSavedPipeline
+};
+
+export default function reducer(state = INITIAL_STATE, action: AnyAction) {
+  const fn = MAPPINGS[action.type];
+  return fn ? fn(state, action) : state;
+}
+
+export const setShowSavedPipelines = (show: boolean) => ({
+  type: SET_SHOW_SAVED_PIPELINES,
+  show
+});
+
+export const savedPipelineAdd = (pipelines: Pipeline[]) => ({
+  type: SAVED_PIPELINE_ADD,
+  pipelines
+});
+
+/**
+ * 
+ * @returns {import('redux').AnyAction}
+ */
+export const getSavedPipelines = (): ThunkAction<void, RootState, void, AnyAction> => 
+  (dispatch, getState) => {
+    if (!getState().savedPipeline.isLoaded) {
+      dispatch(updatePipelineList());
+    }
+  };
+
+/**
+ * Update the pipeline list.
+ *
+ * @returns {Function} The thunk function.
+ */
+export const updatePipelineList = (): ThunkAction<void, RootState, void, AnyAction> =>
+  (dispatch, getState) => {
+    const pipelineStorage = new PipelineStorage();
+    const state = getState();
+    pipelineStorage.loadAll()
+      .then(pipelines => {
+        const thisNamespacePipelines = pipelines.filter(
+          ({namespace}) => namespace === state.namespace
+        );
+        dispatch(setIsModified(false));
+        dispatch(savedPipelineAdd(thisNamespacePipelines));
+        dispatch(globalAppRegistryEmit('agg-pipeline-saved', { name: state.name }));
+      })
+      .catch(err => {
+        debug('Failed to load pipelines', err);
+      });
+  };
+
+/**
+ * Save the current state of your pipeline
+ *
+ * @returns {import('redux').AnyAction} The action.
+ */
+export const saveCurrentPipeline = (): ThunkAction<void, RootState, void, AnyAction> => async (
+  dispatch, getState
+) => {
+  // We dynamically require these libraries as this file is used in cloud and
+  // we don't want global imports of packages not available on the web.
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { promises: fs } = require('fs');
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const path = require('path');
+
+  const state = getState();
+
+  if (getState().id === '') {
+    dispatch(createId());
+  }
+
+  const pipeline = state.pipeline.map((stage) => {
+    return { ...stage, previewDocuments: [] };
+  });
+
+  const {
+    id,
+    name,
+    namespace,
+    comments,
+    sample,
+    autoPreview,
+    collationString: { text },
+    dataService
+  } = getState();
+
+  const stateRecord = {
+    id,
+    name,
+    namespace,
+    comments,
+    sample,
+    autoPreview,
+    collationString: text,
+    pipeline,
+    host:
+      dataService?.dataService?.getConnectionString?.().hosts.join(',') ??
+      null
+  };
+
+  track('Aggregation Saved', {
+    id: stateRecord.id,
+    num_stages: pipeline.length
+  });
+
+  const dirname = getDirectory();
```

Comment:
Not sure if my comment is relevant, but we have [this ticket](https://jira.mongodb.org/browse/COMPASS-5893) to use `pipeline-storage.js` class to handle saving of pipelines in storage. Feel free to ignore this comment

Reply:
Thanks for the heads up on that, will keep that work in the separate ticket/pr.